java/src/jmri/jmrix/DefaultSystemConnectionMemo.java

```diff
@@ -198,9 +198,18 @@ public boolean provides(Class<?> c) {
         }
     }
 
+    // removeRegisteredObject requires a Class<T> argument, but 
+    // the classObjectMap member (and hence keySet here) contains
+    // a Class object.  
+    // It's not at all clear that type-safety is being 
+    // assured, as opposed to assumed, but to avoid warnings on every compile
+    // we'll annotate the assumption.
+    @SuppressWarnings("unchecked")  // foreach statement below assumes type-safety, see comment above.
```

Comment:
If you want to fix the warning change `Class` on lines 34 and 202 (original) 209 (in this PR) to `Class<?>`. The problem here is that the field `classObjectMap` is defined without the use of Generics, while the methods that manipulate it are.

The use of Generics precludes the use of the `instanceof` operator, so any use of Generics anywhere in the code assumes, but does not assure, type safety.

Reply:
Not debating this.  Merge it or not.  A better solution would be even better, but I've contributed this in case it helps somebody else.  I've got it in all my development branches now so as to not have to get past these warnings _every_ _damn_ _time_. If another fix is provided, I'm happy to resolve the resulting conflicts.